plugin/verifier/factory.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package verifier
+
+import "fmt"
+
+// RegisteredVerifiers saves the registered verifier factories.
+var RegisteredVerifiers = make(map[string]VerifierFactory)
+
+// VerifierConfig represents the configuration of a verifier.
+type VerifierConfig struct {
+	Name          string                 `json:"name"`
+	Type          string                 `json:"type"`
```

Comment:
It looks like `Type` is used for registration as the name of a key of a verifier. So why do we need both `Name` and `Type`? Will Type have different values?